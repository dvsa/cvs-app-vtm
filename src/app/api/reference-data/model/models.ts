export * from './reference-data.model';
