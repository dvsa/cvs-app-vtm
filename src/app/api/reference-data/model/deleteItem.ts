/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

export interface DeleteItem { 
    success?: boolean;
}