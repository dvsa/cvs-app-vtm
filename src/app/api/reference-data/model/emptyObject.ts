/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface EmptyObject {
}
