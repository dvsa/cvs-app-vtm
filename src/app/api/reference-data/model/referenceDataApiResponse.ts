/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */
import { ReferenceDataApiResponseWithPagination } from './referenceDataApiResponseWithPagination';
import { ReferenceDataApiResponseWithoutPagination } from './referenceDataApiResponseWithoutPagination';

export type ReferenceDataApiResponse = ReferenceDataApiResponseWithoutPagination | ReferenceDataApiResponseWithPagination;