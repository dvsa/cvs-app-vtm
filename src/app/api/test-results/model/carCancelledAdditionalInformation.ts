/**
 * Test Results Microservice
 * This is the API spec for capturing test results. These test result will be stored in the AWS DynamoDB database. Authorization details will be updated once we have confirmed the security scheme we are using.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */
import { CarCancelledAdditionalInformationLocation } from './carCancelledAdditionalInformationLocation';

export interface CarCancelledAdditionalInformation { 
    location: CarCancelledAdditionalInformationLocation;
    notes: string;
}