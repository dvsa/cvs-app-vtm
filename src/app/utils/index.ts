export * from './mockStore';
export * from './test-model.utils';
export * from './testing.utils';
export * from './form.utils';


