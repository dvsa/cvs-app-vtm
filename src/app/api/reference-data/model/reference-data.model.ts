import { ReferenceDataResourceType } from '@models/reference-data.model';

export type ReferenceDataItem = {
  resourceType: ReferenceDataResourceType;
  resourceKey: string | number;
};

export type ReferenceDataList = Array<ReferenceDataItem>;

export type ReferenceDataApiResponse = {
  data: ReferenceDataList;
};
