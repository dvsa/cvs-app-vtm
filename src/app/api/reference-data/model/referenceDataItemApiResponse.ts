/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */
import { EmptyObject } from './emptyObject';
import { ReferenceDataItem } from './referenceDataItem';

export type ReferenceDataItemApiResponse = ReferenceDataItem | EmptyObject;