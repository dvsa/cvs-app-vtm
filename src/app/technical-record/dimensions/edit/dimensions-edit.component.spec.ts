import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import {
  ReactiveFormsModule,
  FormsModule,
  FormGroupDirective,
  FormGroup,
  AbstractControl
} from '@angular/forms';

import { SharedModule } from '@app/shared';
import { TESTING_UTILS } from '@app/utils/testing.utils';
import { DimensionsEditComponent } from './dimensions-edit.component';
import { TechRecord } from '@app/models/tech-record.model';

describe('DimensionsEditComponent', () => {
  let component: DimensionsEditComponent;
  let fixture: ComponentFixture<DimensionsEditComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [FormsModule, ReactiveFormsModule, SharedModule],
      declarations: [DimensionsEditComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
      providers: [
        FormGroupDirective,
        {
          provide: FormGroupDirective,
          useValue: TESTING_UTILS.mockFormGroupDirective({
            techRecord: new FormGroup({}) as AbstractControl
          })
        }
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DimensionsEditComponent);
    component = fixture.componentInstance;
  });

  it('should create with initialized form dimensions controls', () => {
    component.techRecord = {
      frontAxleTo5thWheelMin: 456,
      frontAxleTo5thWheelMax: 660,
      frontAxleTo5thWheelCouplingMin: 100,
      frontAxleTo5thWheelCouplingMax: 150,
      dimensions: TESTING_UTILS.mockDimensions()
    } as TechRecord;
    fixture.detectChanges();

    expect(component).toBeDefined();
    expect(fixture).toMatchSnapshot();
  });

  // TODO: CVSB-10619 test for AXLE Dimension combination control;
});
