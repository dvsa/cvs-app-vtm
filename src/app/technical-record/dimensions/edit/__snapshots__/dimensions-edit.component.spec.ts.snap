// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DimensionsEditComponent should create with initialized form dimensions controls 1`] = `
<vtm-dimensions-edit
  detectChange={[Function ViewRef_]}
  fb={[Function FormBuilder]}
  numberOfAxles$={[Function Observable]}
  onDestroy$={[Function Subject]}
  parent={[Function FormGroupDirective]}
  techRecHelper={[Function TechRecordHelperService]}
  techRecord={[Function Object]}
  techRecordFg={[Function FormGroup]}
>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-three-quarters"
    >
      <div
        class="govuk-form-group"
      >
        <fieldset
          class="govuk-fieldset"
        >
          <form
            class="ng-untouched ng-pristine ng-valid"
            ng-reflect-form="[object Object]"
            novalidate=""
          >
            
            <div
              class="govuk-form-group ng-untouched ng-pristine ng-valid"
              ng-reflect-form="[object Object]"
            >
              <label
                class="govuk-label govuk-!-font-weight-bold"
                for="test-edit-length"
              >
                 Length (mm) 
              </label>
              <input
                class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                id="test-edit-length"
                maxlength="5"
                name="length"
                ng-reflect-maxlength="5"
                ng-reflect-name="length"
                type="number"
              />
            </div>
            
            <div
              class="govuk-form-group ng-untouched ng-pristine ng-valid"
              ng-reflect-form="[object Object]"
            >
              <label
                class="govuk-label govuk-!-font-weight-bold"
                for="test-edit-width"
              >
                 Width (mm) 
              </label>
              <input
                class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                id="test-edit-width"
                maxlength="5"
                name="width"
                ng-reflect-maxlength="5"
                ng-reflect-name="width"
                type="number"
              />
            </div>
            
            <div
              class="govuk-form-group ng-untouched ng-pristine ng-valid"
              ng-reflect-form="[object Object]"
            >
              <label
                class="govuk-label govuk-!-font-weight-bold"
                for="test-edit-frontAxleToRearAxle"
              >
                 Front axle to rear axle (mm) 
              </label>
              <input
                class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                id="test-edit-frontAxleToRearAxle"
                maxlength="5"
                name="frontAxleToRearAxle"
                ng-reflect-maxlength="5"
                ng-reflect-name="frontAxleToRearAxle"
                type="number"
              />
            </div>
            <label
              class="govuk-label govuk-!-font-weight-bold"
            >
               Front of vehicle to 5th wheel coupling 
            </label>
            <div
              class="govuk-inset-text govuk-!-margin-top-3"
            >
              
              
              <div
                class="govuk-form-group govuk-!-margin-left-3"
              >
                
                <div
                  class="govuk-form-group ng-untouched ng-pristine ng-valid"
                  ng-reflect-form="[object Object]"
                >
                  <label
                    class="govuk-label govuk-!-font-weight-bold"
                    for="test-edit-frontAxleTo5thWheelCouplingMin"
                  >
                     Minimum (mm) 
                  </label>
                  <input
                    class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                    id="test-edit-frontAxleTo5thWheelCouplingMin"
                    maxlength="5"
                    name="frontAxleTo5thWheelCouplingMin"
                    ng-reflect-maxlength="5"
                    ng-reflect-name="frontAxleTo5thWheelCouplingMin"
                    type="number"
                  />
                </div>
              </div>
              
              <div
                class="govuk-form-group govuk-!-margin-left-3"
              >
                
                <div
                  class="govuk-form-group ng-untouched ng-pristine ng-valid"
                  ng-reflect-form="[object Object]"
                >
                  <label
                    class="govuk-label govuk-!-font-weight-bold"
                    for="test-edit-frontAxleTo5thWheelCouplingMax"
                  >
                     Maximum (mm) 
                  </label>
                  <input
                    class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                    id="test-edit-frontAxleTo5thWheelCouplingMax"
                    maxlength="5"
                    name="frontAxleTo5thWheelCouplingMax"
                    ng-reflect-maxlength="5"
                    ng-reflect-name="frontAxleTo5thWheelCouplingMax"
                    type="number"
                  />
                </div>
              </div>
            </div>
            <label
              class="govuk-label govuk-!-font-weight-bold"
            >
               Front axle to 5th wheel 
            </label>
            <div
              class="govuk-inset-text govuk-!-margin-top-3"
            >
              
              
              <div
                class="govuk-form-group govuk-!-margin-left-3"
              >
                
                <div
                  class="govuk-form-group ng-untouched ng-pristine ng-valid"
                  ng-reflect-form="[object Object]"
                >
                  <label
                    class="govuk-label govuk-!-font-weight-bold"
                    for="test-edit-frontAxleTo5thWheelMin"
                  >
                     Minimum (mm) 
                  </label>
                  <input
                    class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                    id="test-edit-frontAxleTo5thWheelMin"
                    maxlength="5"
                    name="frontAxleTo5thWheelMin"
                    ng-reflect-maxlength="5"
                    ng-reflect-name="frontAxleTo5thWheelMin"
                    type="number"
                  />
                </div>
              </div>
              
              <div
                class="govuk-form-group govuk-!-margin-left-3"
              >
                
                <div
                  class="govuk-form-group ng-untouched ng-pristine ng-valid"
                  ng-reflect-form="[object Object]"
                >
                  <label
                    class="govuk-label govuk-!-font-weight-bold"
                    for="test-edit-frontAxleTo5thWheelMax"
                  >
                     Maximum (mm) 
                  </label>
                  <input
                    class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                    id="test-edit-frontAxleTo5thWheelMax"
                    maxlength="5"
                    name="frontAxleTo5thWheelMax"
                    ng-reflect-maxlength="5"
                    ng-reflect-name="frontAxleTo5thWheelMax"
                    type="number"
                  />
                </div>
              </div>
            </div>
            
            
            
          </form>
        </fieldset>
      </div>
    </div>
  </div>
</vtm-dimensions-edit>
`;

exports[`DimensionsEditComponent should render AXLE SPACING controls on initialization 1`] = `
<vtm-dimensions-edit
  detectChange={[Function ViewRef_]}
  fb={[Function FormBuilder]}
  numberOfAxles$={[Function Observable]}
  onDestroy$={[Function Subject]}
  parent={[Function FormGroupDirective]}
  techRecHelper={[Function TechRecordHelperService]}
  techRecord={[Function Object]}
  techRecordFg={[Function FormGroup]}
>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-three-quarters"
    >
      <div
        class="govuk-form-group"
      >
        <fieldset
          class="govuk-fieldset"
        >
          <form
            class="ng-untouched ng-pristine ng-valid"
            ng-reflect-form="[object Object]"
            novalidate=""
          >
            
            <div
              class="govuk-form-group ng-untouched ng-pristine ng-valid"
              ng-reflect-form="[object Object]"
            >
              <label
                class="govuk-label govuk-!-font-weight-bold"
                for="test-edit-length"
              >
                 Length (mm) 
              </label>
              <input
                class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                id="test-edit-length"
                maxlength="5"
                name="length"
                ng-reflect-maxlength="5"
                ng-reflect-name="length"
                type="number"
              />
            </div>
            
            <div
              class="govuk-form-group ng-untouched ng-pristine ng-valid"
              ng-reflect-form="[object Object]"
            >
              <label
                class="govuk-label govuk-!-font-weight-bold"
                for="test-edit-width"
              >
                 Width (mm) 
              </label>
              <input
                class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                id="test-edit-width"
                maxlength="5"
                name="width"
                ng-reflect-maxlength="5"
                ng-reflect-name="width"
                type="number"
              />
            </div>
            
            <div
              class="govuk-form-group ng-untouched ng-pristine ng-valid"
              ng-reflect-form="[object Object]"
            >
              <label
                class="govuk-label govuk-!-font-weight-bold"
                for="test-edit-frontAxleToRearAxle"
              >
                 Front axle to rear axle (mm) 
              </label>
              <input
                class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                id="test-edit-frontAxleToRearAxle"
                maxlength="5"
                name="frontAxleToRearAxle"
                ng-reflect-maxlength="5"
                ng-reflect-name="frontAxleToRearAxle"
                type="number"
              />
            </div>
            <label
              class="govuk-label govuk-!-font-weight-bold"
            >
               Front of vehicle to 5th wheel coupling 
            </label>
            <div
              class="govuk-inset-text govuk-!-margin-top-3"
            >
              
              
              <div
                class="govuk-form-group govuk-!-margin-left-3"
              >
                
                <div
                  class="govuk-form-group ng-untouched ng-pristine ng-valid"
                  ng-reflect-form="[object Object]"
                >
                  <label
                    class="govuk-label govuk-!-font-weight-bold"
                    for="test-edit-frontAxleTo5thWheelCouplingMin"
                  >
                     Minimum (mm) 
                  </label>
                  <input
                    class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                    id="test-edit-frontAxleTo5thWheelCouplingMin"
                    maxlength="5"
                    name="frontAxleTo5thWheelCouplingMin"
                    ng-reflect-maxlength="5"
                    ng-reflect-name="frontAxleTo5thWheelCouplingMin"
                    type="number"
                  />
                </div>
              </div>
              
              <div
                class="govuk-form-group govuk-!-margin-left-3"
              >
                
                <div
                  class="govuk-form-group ng-untouched ng-pristine ng-valid"
                  ng-reflect-form="[object Object]"
                >
                  <label
                    class="govuk-label govuk-!-font-weight-bold"
                    for="test-edit-frontAxleTo5thWheelCouplingMax"
                  >
                     Maximum (mm) 
                  </label>
                  <input
                    class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                    id="test-edit-frontAxleTo5thWheelCouplingMax"
                    maxlength="5"
                    name="frontAxleTo5thWheelCouplingMax"
                    ng-reflect-maxlength="5"
                    ng-reflect-name="frontAxleTo5thWheelCouplingMax"
                    type="number"
                  />
                </div>
              </div>
            </div>
            <label
              class="govuk-label govuk-!-font-weight-bold"
            >
               Front axle to 5th wheel 
            </label>
            <div
              class="govuk-inset-text govuk-!-margin-top-3"
            >
              
              
              <div
                class="govuk-form-group govuk-!-margin-left-3"
              >
                
                <div
                  class="govuk-form-group ng-untouched ng-pristine ng-valid"
                  ng-reflect-form="[object Object]"
                >
                  <label
                    class="govuk-label govuk-!-font-weight-bold"
                    for="test-edit-frontAxleTo5thWheelMin"
                  >
                     Minimum (mm) 
                  </label>
                  <input
                    class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                    id="test-edit-frontAxleTo5thWheelMin"
                    maxlength="5"
                    name="frontAxleTo5thWheelMin"
                    ng-reflect-maxlength="5"
                    ng-reflect-name="frontAxleTo5thWheelMin"
                    type="number"
                  />
                </div>
              </div>
              
              <div
                class="govuk-form-group govuk-!-margin-left-3"
              >
                
                <div
                  class="govuk-form-group ng-untouched ng-pristine ng-valid"
                  ng-reflect-form="[object Object]"
                >
                  <label
                    class="govuk-label govuk-!-font-weight-bold"
                    for="test-edit-frontAxleTo5thWheelMax"
                  >
                     Maximum (mm) 
                  </label>
                  <input
                    class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                    id="test-edit-frontAxleTo5thWheelMax"
                    maxlength="5"
                    name="frontAxleTo5thWheelMax"
                    ng-reflect-maxlength="5"
                    ng-reflect-name="frontAxleTo5thWheelMax"
                    type="number"
                  />
                </div>
              </div>
            </div>
            
            
            <div
              class="ng-untouched ng-pristine ng-valid"
              formarrayname="axleSpacing"
              ng-reflect-name="axleSpacing"
            >
              
              
              
              <div
                class="govuk-form-group ng-untouched ng-pristine ng-valid"
                ng-reflect-form="[object Object]"
              >
                <label
                  class="govuk-label govuk-!-font-weight-bold"
                  for="test-edit-axleSpacing-value-1"
                >
                   Axle 1 to axle 2 (mm) 
                </label>
                <input
                  class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                  id="test-edit-axleSpacing-value-1"
                  maxlength="5"
                  name="value"
                  ng-reflect-maxlength="5"
                  ng-reflect-name="value"
                  type="number"
                />
              </div>
              
              
              
              <div
                class="govuk-form-group ng-untouched ng-pristine ng-valid"
                ng-reflect-form="[object Object]"
              >
                <label
                  class="govuk-label govuk-!-font-weight-bold"
                  for="test-edit-axleSpacing-value-2"
                >
                   Axle 2 to axle 3 (mm) 
                </label>
                <input
                  class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
                  id="test-edit-axleSpacing-value-2"
                  maxlength="5"
                  name="value"
                  ng-reflect-maxlength="5"
                  ng-reflect-name="value"
                  type="number"
                />
              </div>
              
            </div>
            
          </form>
        </fieldset>
      </div>
    </div>
  </div>
</vtm-dimensions-edit>
`;
